projects/plugins/jetpack/class.jetpack-connection-banner.php

```diff
@@ -75,16 +75,30 @@ function build_connect_url_for_slide( $jp_version_banner_added ) {
 	 * Will initialize hooks to display the new (as of 4.4) connection banner if the current user can
 	 * connect Jetpack, if Jetpack has not been deactivated, and if the current page is the plugins page.
 	 *
-	 * This method should not be called if the site is connected to WordPress.com or if the site is in offline mode.
-	 *
 	 * @since 4.4.0
 	 * @since 4.5.0 Made the new (as of 4.4) connection banner display to everyone by default.
 	 * @since 5.3.0 Running another split test between 4.4 banner and a new one in 5.3.
 	 * @since 7.2   B test was removed.
+	 * @since 9.7   Moved the connection condition checking to this method to fulfill Licensing requirements.
 	 *
 	 * @param $current_screen
 	 */
 	function maybe_initialize_hooks( $current_screen ) {
+		$has_connected_owner = Jetpack::connection()->has_connected_owner();
+		$is_connected        = ( new Status() )->is_no_user_testing_mode() ? Jetpack::connection()->is_connected() : $has_connected_owner;
```

Comment:
Could this be updated to use `Jetpack::is_connection_ready()`?

cc @leogermani